compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1615,6 +1615,10 @@ rustc_queries! {
     query all_traits(_: ()) -> &'tcx [DefId] {
         desc { "fetching all foreign and local traits" }
     }
+    query all_traits2(cnum: CrateNum) -> &'tcx [DefId] {
```

Comment:
What about `all_traits_of_crate`? The current name is unclear.

Reply:
This PR is WIP, implemented just enough to run benchmarks, I'll cleanup the code later today.